tests/run-pass/file_read.rs

```diff
@@ -2,12 +2,22 @@
 // compile-flags: -Zmiri-disable-isolation
 
 use std::fs::File;
-use std::io::Read;
+use std::io::{ Read, Write };
 
 fn main() {
-    // FIXME: create the file and delete it when `rm` is implemented.
-    let mut file = File::open("./tests/hello.txt").unwrap();
-    let mut contents = String::new();
-    file.read_to_string(&mut contents).unwrap();
-    assert_eq!("Hello, World!\n", contents);
+    // FIXME: remove the file and delete it when `rm` is implemented.
+    let path = "./tests/hello.txt";
+    let bytes = b"Hello, World!\n";
+    // Test creating, writing and closing a file (closing is tested when `file` is dropped).
+    let mut file = File::create(path).unwrap();
+    file.write(bytes).unwrap();
+    // Test opening, reading and closing a file.
+    let mut file = File::open(path).unwrap();
+    let mut contents = Vec::new();
+    // Reading 0 bytes should not fill `contents`.
```

Comment:
heh, it won't ever do that, `read` takes a `&mut [u8]`, and an empty `Vec` is just coerced to a zero length slice. There's no logic here that could possibly resize the `Vec`.

Reply:
but then how do we test that `read` didn't write anything to the buffer?

Reply:
well.. if it did read any bytes, then the next read would not get all bytes, thus your assertion for the content later would fail